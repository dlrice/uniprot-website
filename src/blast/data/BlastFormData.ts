export type BlastFormValue = {
  fieldName: string;
  selected?: string | undefined;
  selectedLabel?: string | undefined;
  type: BlastFieldTypes;
  values?: { label?: string; value?: string | number }[];
};

export enum BlastFields {
  program = 'Program',
  stype = 'Sequence type',
  email = 'Email',
  sequence = 'Sequence',
  taxon = 'Taxon',
  targetDb = 'Target database',
  threshold = 'E-Threshold',
  matrix = 'Matrix',
  filter = 'Filter',
  gapped = 'Gapped',
  hits = 'Hits',
}

export enum BlastFieldTypes {
  textarea,
  select,
  autocomplete,
}

export type BlastFormValues = { [x in BlastFields]: BlastFormValue };

export default {
  [BlastFields.program]: {
    fieldName: 'program',
    selected: 'blastp',
  },
  [BlastFields.stype]: {
    fieldName: 'stype',
    selected: 'protein',
  },
  [BlastFields.email]: {
    fieldName: 'email',
    selected: 'uuw_dev@uniprot.org',
  },
  [BlastFields.sequence]: {
    fieldName: 'sequence',
    type: BlastFieldTypes.textarea,
  },
  [BlastFields.targetDb]: {
    fieldName: 'database',
    type: BlastFieldTypes.select,
    selected: 'uniprotkb_refprotswissprot',
    values: [
      {
        value: 'uniprotkb_refprotswissprot',
        label: 'UniProtKb',
      },
    ],
  },
  [BlastFields.taxon]: {
    fieldName: 'taxids',
    type: BlastFieldTypes.autocomplete,
  },
  [BlastFields.threshold]: {
    fieldName: 'threshold',
    type: BlastFieldTypes.select,
    selected: '10',
    values: [
      { value: '0.0001' },
      { value: '0.001' },
      { value: '0.01' },
      { value: '0.1' },
      { value: '1' },
      { value: '10' },
      { value: '100' },
      { value: '1000' },
    ],
  },
  [BlastFields.matrix]: {
    fieldName: 'matrix',
    type: BlastFieldTypes.select,
    selected: null,
    values: [
      {
        value: '',
        label: 'Auto',
      },
      { value: '45' },
      { value: '62' },
      { value: '80' },
      { value: '70' },
      { value: '30' },
    ],
  },
  [BlastFields.filter]: {
    fieldName: 'filter',
    type: BlastFieldTypes.select,
    selected: 'F',
    values: [
      { value: 'F', label: 'None' },
      { valueNone: 'T', label: 'Filter low complexity regions' },
      { value: 'mask', label: 'Mask lookup table only' },
    ],
  },
  [BlastFields.gapped]: {
    fieldName: 'gapped',
    type: BlastFieldTypes.select,
    selected: 'true',
    values: [
      { value: 'true', label: 'yes' },
      { value: 'false', label: 'no' },
    ],
  },
  [BlastFields.hits]: {
    fieldName: 'hits',
    type: BlastFieldTypes.select,
    selected: '250',
    values: [
      { value: '50' },
      { value: '100' },
      { value: '250' },
      { value: '500' },
      { value: '750' },
      { value: '1000' },
    ],
  },
};
