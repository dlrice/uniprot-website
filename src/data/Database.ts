import { Database, DatabaseCategory } from '../model/types/databaseTypes';
import EntrySection from '../model/types/EntrySection';

export const databaseCategoryToString = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

const databaseCategoryToDatabases = new Map<DatabaseCategory, Database[]>();
databaseCategoryToDatabases.set(DatabaseCategory.SEQUENCE, [
  // Not found: EMBL/GenBank/DDBJ
  Database.EMBL,
  Database.CCDS,
  Database.PIR,
  Database.RefSeq,
  Database.UniGene,
]);
databaseCategoryToDatabases.set(DatabaseCategory.STRUCTURE, [
  Database.PDB,
  Database.SMR,
  Database.DisProt,
  Database.ProteinModelPortal,
]);
databaseCategoryToDatabases.set(DatabaseCategory.INTERACTION, [
  Database.BioGrid,
  Database.DIP,
  Database.IntAct,
  Database.MINT,
  Database.STRING,
]);
databaseCategoryToDatabases.set(DatabaseCategory.CHEMISTRY, [
  Database.BindingDB,
  Database.ChEMBL,
  Database.DrugBank,
  Database.GuidetoPHARMACOLOGY,
]);
databaseCategoryToDatabases.set(DatabaseCategory.FAMILY, [
  // Not found: Database.PptaseDB,
  Database.Allergome,
  Database.CAZy,
  Database.MEROPS,
  Database.MoonProt,
  Database.mycoCLAP,
  Database.PeroxiBase,
  Database.REBASE,
  Database.TCDB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PTM, [
  // Not found: DEPOD, PhosphoSite, PhosSite
  Database.DEPOD,
  Database.PhosphoSitePlus,
  Database.UniCarbKB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.POLYMORPHISM, [
  Database.dbSNP,
  Database.DMDM,
]);
databaseCategoryToDatabases.set(DatabaseCategory.GEL, [
  Database.SWISS_2DPAGE,
  Database.COMPLUYEAST_2DPAGE,
  Database.DOSAC_COBS_2DPAGE,
  Database.OGP,
  Database.REPRODUCTION_2DPAGE,
  Database.UCD_2DPAGE,
  Database.World_2DPAGE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTEOMIC, [
  Database.MaxQB,
  Database.PaxDb,
  Database.PeptideAtlas,
  Database.PRIDE,
  Database.ProMEX,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTOCOL, [Database.DNASU]);
databaseCategoryToDatabases.set(DatabaseCategory.GENOME, [
  Database.Ensembl,
  Database.EnsemblBacteria,
  Database.EnsemblFungi,
  Database.EnsemblMetazoa,
  Database.EnsemblPlants,
  Database.EnsemblProtists,
  Database.GeneCards,
  Database.GeneID,
  Database.KEGG,
  Database.PATRIC,
  Database.UCSC,
  Database.VectorBase,
]);
databaseCategoryToDatabases.set(DatabaseCategory.ORGANISM, [
  Database.ArachnoServer,
  Database.CGD,
  Database.ConoServer,
  Database.CTD,
  Database.dictyBase,
  Database.EchoBASE,
  Database.EcoGene,
  Database.euHCVdb,
  Database.EuPathDB,
  Database.FlyBase,
  Database.GeneReviews,
  Database.Gramene,
  Database.H_InvDB,
  Database.HGNC,
  Database.HPA,
  Database.LegioList,
  Database.Leproma,
  Database.MaizeGDB,
  Database.MIM,
  Database.MGI,
  Database.neXtProt,
  Database.Orphanet,
  Database.PharmGKB,
  Database.PomBase,
  Database.PseudoCAP,
  Database.RGD,
  Database.SGD,
  Database.TAIR,
  Database.TubercuList,
  Database.WormBase,
  Database.Xenbase,
  Database.ZFIN,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PHYLOGENOMIC, [
  Database.eggNOG,
  Database.GeneTree,
  Database.HOGENOM,
  Database.HOVERGEN,
  Database.InParanoid,
  Database.KO,
  Database.OMA,
  Database.OrthoDB,
  Database.PhylomeDB,
  Database.TreeFam,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PATHWAY, [
  Database.BioCyc,
  Database.BRENDA,
  Database.Reactome,
  Database.SABIO_RK,
  Database.SignaLink,
  Database.UniPathway,
]);
databaseCategoryToDatabases.set(DatabaseCategory.EXPRESSION, [
  Database.Bgee,
  Database.ExpressionAtlas,
]);
databaseCategoryToDatabases.set(DatabaseCategory.DOMAIN, [
  Database.HAMAP,
  Database.InterPro,
  Database.Gene3D,
  Database.PANTHER,
  Database.Pfam,
  Database.PIRSF,
  Database.PRINTS,
  Database.ProDom,
  Database.SMART,
  Database.SUPFAM,
  Database.TIGRFAMs,
  Database.PROSITE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.OTHER, [
  // Not found: NextBio
  Database.ChiTaRS,
  Database.EvolutionaryTrace,
  Database.GenomeRNAi,
  Database.GeneWiki,
  Database.PMAP_CutDB,
  Database.PRO,
]);

const entrySectionToDatabaseCategories = new Map<
  EntrySection,
  DatabaseCategory[]
>();
entrySectionToDatabaseCategories.set(EntrySection.Expression, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.EXPRESSION,
]);
entrySectionToDatabaseCategories.set(EntrySection.FamilyAndDomains, [
  DatabaseCategory.FAMILY,
  DatabaseCategory.PHYLOGENOMIC,
  DatabaseCategory.DOMAIN,
]);
entrySectionToDatabaseCategories.set(EntrySection.Function, [
  DatabaseCategory.PATHWAY,
  DatabaseCategory.FAMILY,
]);
entrySectionToDatabaseCategories.set(EntrySection.Interaction, [
  DatabaseCategory.INTERACTION,
  DatabaseCategory.CHEMISTRY,
]);
entrySectionToDatabaseCategories.set(EntrySection.NamesAndTaxonomy, [
  DatabaseCategory.ORGANISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.PathologyOrBioTech, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.OTHER,
  DatabaseCategory.FAMILY,
  DatabaseCategory.CHEMISTRY,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.ProteinProcessing, [
  DatabaseCategory.PROTEOMIC,
  DatabaseCategory.GEL,
  DatabaseCategory.PTM,
  DatabaseCategory.OTHER,
]);
entrySectionToDatabaseCategories.set(EntrySection.Sequence, [
  DatabaseCategory.SEQUENCE,
  DatabaseCategory.GENOME,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.Structure, [
  DatabaseCategory.STRUCTURE,
  DatabaseCategory.PROTOCOL,
  DatabaseCategory.OTHER,
]);

// Combine multiple maps into single map for quick read access.
export const entrySectionDatabaseToDatabaseCategory = new Map<
  string,
  DatabaseCategory
>();
for (const [
  entrySection,
  databaseCategories,
] of entrySectionToDatabaseCategories) {
  if (databaseCategories) {
    for (const databaseCategory of databaseCategories) {
      const databases = databaseCategoryToDatabases.get(databaseCategory);
      if (databases) {
        for (const database of databases) {
          entrySectionDatabaseToDatabaseCategory.set(
            `${entrySection}_${database}`,
            databaseCategory
          );
        }
      }
    }
  }
}

export default entrySectionDatabaseToDatabaseCategory;
