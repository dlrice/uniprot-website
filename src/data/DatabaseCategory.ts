enum databaseCategory {
  SEQUENCE = 'SEQUENCE',
  STRUCTURE = 'STRUCTURE',
  INTERACTION = 'INTERACTION',
  CHEMISTRY = 'CHEMISTRY',
  FAMILY = 'FAMILY',
  PTM = 'PTM',
  POLYMORPHISM = 'POLYMORPHISM',
  GEL = 'GEL',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PROTOCOL',
  GENOME = 'GENOME',
  ORGANISM = 'ORGANISM',
  PHYLOGENOMIC = 'PHYLOGENOMIC',
  PATHWAY = 'PATHWAY',
  EXPRESSION = 'EXPRESSION',
  DOMAIN = 'DOMAIN',
  OTHER = 'OTHER',
}

export default databaseCategory;
