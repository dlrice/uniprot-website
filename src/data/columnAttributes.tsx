const small = 200; //10rem;
const medium = 400; //15rem;
const large = 600; //20rem;

export type columnAttributes = {
  [columnName: string]: {
    width: number;
    label: string;
  };
};

const columnAttributes: columnAttributes = {
  accession: {
    width: small,
    label: 'Entry',
  },
  id: {
    width: small,
    label: 'Entry Name',
  },
  protein_name: {
    width: large,
    label: 'Protein names',
  },
  gene_names: {
    width: medium,
    label: 'Gene Names',
  },
  organism: {
    width: large,
    label: 'Organism',
  },
};

export default columnAttributes;
