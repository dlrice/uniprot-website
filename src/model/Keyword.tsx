import React, { Fragment } from 'react';
import { v1 } from 'uuid';
import { InfoList } from 'franklin-sites';
import { Link } from 'react-router-dom';
import EntrySectionType from './types/EntrySection';
import {
  entrySectionToKeywordCategories,
  KeywordCategory,
} from './types/keywordTypes';

type Keyword = {
  id?: string;
  value?: string;
  category?: KeywordCategory;
};

type Data = {
  keywords?: Keyword[];
};

type KeywordProps = {
  section: EntrySectionType;
  data: Data;
};

type KeywordListProps = {
  keywords?: Keyword[];
};

type KeywordItempProps = {
  id?: string;
  value?: string;
};

export const KeywordItem: React.FC<KeywordItempProps> = ({ id, value }) => {
  if (!id || !value) {
    return null;
  }
  return <Link to={`/keywords/${id}`}>{value}</Link>;
};

export const KeywordList: React.FC<KeywordListProps> = ({ keywords }) => {
  if (!keywords) {
    return null;
  }
  const nodes = keywords.map((keyword, index) => {
    const { id, value } = keyword;
    if (!id || !value) {
      return null;
    }
    return (
      <Fragment key={v1()}>
        <KeywordItem id={id} value={value} />
        {index < keywords.length - 1 && ', '}
      </Fragment>
    );
  });

  return <Fragment>{nodes}</Fragment>;
};

export const Keyword: React.FC<KeywordProps> = ({ data, section }) => {
  const { keywords } = data;
  const keywordCategories = entrySectionToKeywordCategories.get(section);
  if (!keywords || !keywordCategories) {
    return null;
  }
  const foundCategoryToKeywords: { [keywordCategory: string]: Keyword[] } = {};
  for (const keyword of keywords) {
    const { category, value, id } = keyword;
    if (!category || !value || !id || !keywordCategories.includes(category)) {
      continue;
    }
    if (category in foundCategoryToKeywords) {
      foundCategoryToKeywords[category].push(keyword);
    } else {
      foundCategoryToKeywords[category] = [keyword];
    }
  }
  const infoData = Object.keys(foundCategoryToKeywords)
    .sort()
    .map(foundCategory => ({
      title: foundCategory,
      content: (
        <KeywordList keywords={foundCategoryToKeywords[foundCategory]} />
      ),
    }));
  const node = infoData.length && (
    <Fragment>
      <h4>Keywords</h4>
      <InfoList infoData={infoData} />
    </Fragment>
  );
  return <Fragment>{node}</Fragment>;
};

export default Keyword;
