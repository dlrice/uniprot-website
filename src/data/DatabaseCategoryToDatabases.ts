import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string[]>([
  [
    DatabaseCategory.SEQUENCE,
    ['EMBL/GenBank/DDBJ', 'CCDS', 'PIR', 'RefSeq', 'UniGene'],
  ],
  [DatabaseCategory.STRUCTURE, ['PDB', 'SMR', 'DisProt', 'ProteinModelPortal']],
  [
    DatabaseCategory.INTERACTION,
    ['BioGrid', 'DIP', 'IntAct', 'MINT', 'STRING'],
  ],
  [
    DatabaseCategory.CHEMISTRY,
    ['BindingDB', 'ChEMBL', 'DrugBank', 'GuidetoPHARMACOLOGY'],
  ],
  [
    DatabaseCategory.FAMILY,
    [
      'Allergome',
      'CAZy',
      'MEROPS',
      'MoonProt',
      'mycoCLAP',
      'PeroxiBase',
      'PptaseDB',
      'REBASE',
      'TCDB',
    ],
  ],
  [DatabaseCategory.PTM, ['DEPOD', 'PhosphoSite', 'PhosSite', 'UniCarbKB']],
  [DatabaseCategory.POLYMORPHISM, ['dbSNP', 'DMDM']],
  [
    DatabaseCategory.GEL,
    [
      'SWISS-2DPAGE',
      'COMPLUYEAST-2DPAGE',
      'DOSAC-COBS-2DPAGE',
      'OGP',
      'REPRODUCTION-2DPAGE',
      'UCD-2DPAGE',
      'World-2DPAGE',
    ],
  ],
  [
    DatabaseCategory.PROTEOMIC,
    ['MaxQB', 'PaxDb', 'PeptideAtlas', 'PRIDE', 'ProMEX'],
  ],
  [DatabaseCategory.PROTOCOL, ['DNASU']],
  [
    DatabaseCategory.GENOME,
    [
      'Ensembl and Ensembl Genomes',
      'Ensembl',
      'EnsemblBacteria',
      'EnsemblFungi',
      'EnsemblMetazoa',
      'EnsemblPlants',
      'EnsemblProtists',
      'GeneCards',
      'GeneID',
      'KEGG',
      'PATRIC',
      'UCSC',
      'VectorBase',
    ],
  ],
  [
    DatabaseCategory.ORGANISM,
    [
      'ArachnoServer',
      'CGD',
      'ConoServer',
      'CTD',
      'dictyBase',
      'EchoBASE',
      'EcoGene',
      'euHCVdb',
      'EuPathDB',
      'FlyBase',
      'GeneReviews',
      'Gramene',
      'H-InvDB',
      'HGNC',
      'HPA',
      'LegioList',
      'Leproma',
      'MaizeGDB',
      'MIM',
      'MGI',
      'neXtProt',
      'Orphanet',
      'PharmGKB',
      'PomBase',
      'PseudoCAP',
      'RGD',
      'SGD',
      'TAIR',
      'TubercuList',
      'WormBase',
      'Xenbase',
      'ZFIN',
    ],
  ],
  [
    DatabaseCategory.PHYLOGENOMIC,
    [
      'eggNOG',
      'GeneTree',
      'HOGENOM',
      'HOVERGEN',
      'InParanoid',
      'KO',
      'OMA',
      'OrthoDB',
      'PhylomeDB',
      'TreeFam',
    ],
  ],
  [
    DatabaseCategory.PATHWAY,
    ['BioCyc', 'BRENDA', 'Reactome', 'SABIO-RK', 'SignaLink', 'UniPathway'],
  ],
  [DatabaseCategory.EXPRESSION, ['Bgee', 'ExpressionAtlas']],
  [
    DatabaseCategory.DOMAIN,
    [
      'HAMAP',
      'InterPro',
      'Gene3D',
      'PANTHER',
      'Pfam',
      'PIRSF',
      'PRINTS',
      'ProDom',
      'SMART',
      'SUPFAM',
      'TIGRFAMs',
      'PROSITE',
    ],
  ],
  [
    DatabaseCategory.OTHER,
    [
      'ChiTaRS',
      'EvolutionaryTrace',
      'GenomeRNAi',
      'GeneWiki',
      'NextBio',
      'PMAP-CutDB',
      'PRO',
    ],
  ],
]);

export default databaseCategoryToDatabases;
