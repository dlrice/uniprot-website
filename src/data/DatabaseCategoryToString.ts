import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

export default databaseCategoryToDatabases;
