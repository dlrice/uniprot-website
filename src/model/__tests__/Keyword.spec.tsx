import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { KeywordItem, KeywordList, Keyword } from '../Keyword';
import data from './modelData.json';
import EntrySectionType from '../types/EntrySection';

configure({ adapter: new Adapter() });

describe('Keyword', () => {
  for (const section in EntrySectionType) {
    test(`should render for ${section} section`, () => {
      const wrapper = shallow(
        <Keyword data={data} section={section as EntrySectionType} />
      );
      expect(wrapper.debug()).toMatchSnapshot();
    });
  }
});

describe('KeywordList', () => {
  test('should render for database category GENOME and Ensembl', () => {
    const { keywords } = data;
    const wrapper = shallow(<KeywordList keywords={keywords} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});

describe('KeywordItem', () => {
  const { keywords } = data;
  for (const keyword of keywords) {
    test('should render for database Ensembl', () => {
      const wrapper = shallow(
        <KeywordItem id={keyword.id} value={keyword.value} />
      );
      expect(wrapper.debug()).toMatchSnapshot();
    });
  }
});
