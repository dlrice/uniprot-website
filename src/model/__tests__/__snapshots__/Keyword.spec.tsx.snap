// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyword should render for Expression section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`Keyword should render for FamilyAndDomains section 1`] = `
"<Fragment>
  <h4>
    Keywords
  </h4>
  <Component infoData={{...}} />
</Fragment>"
`;

exports[`Keyword should render for Function section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`Keyword should render for Interaction section 1`] = `""`;

exports[`Keyword should render for Miscellaneous section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`Keyword should render for NamesAndTaxonomy section 1`] = `""`;

exports[`Keyword should render for PathologyOrBioTech section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`Keyword should render for ProteinProcessing section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`Keyword should render for Sequence section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`Keyword should render for Structure section 1`] = `""`;

exports[`Keyword should render for SubCellularLocation section 1`] = `
"<Fragment>
  0
</Fragment>"
`;

exports[`KeywordItem should render for database Ensembl 1`] = `
"<a href=\\"https://www.uniprot.org/keywords/KW-11111\\">
  keyword value
</a>"
`;

exports[`KeywordList should render for database category GENOME and Ensembl 1`] = `
"<Fragment>
  <KeywordItem id=\\"KW-11111\\" value=\\"keyword value\\" />
</Fragment>"
`;
